PhabricatorBMOAuthAdapter.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+final class PhabricatorBMOAuthAdapter extends PhutilAuthAdapter {
+
+  private $account_id;
+  private $account_email;
+  private $account_name;
+  private $account_real_name;
+  private $auth_uri;
+
+  private $type;
+  private $domain;
+
+  public function getAccountID() {
+    return $this->account_id;
+  }
+
+  public function setAccountId($id) {
+    $this->account_id = $id;
+  }
+
+  public function getAdapterType() {
+    return $this->type;
+  }
+
+  public function setAdapterType($type) {
+    $this->type = $type;
+  }
+
+  public function getAdapterDomain() {
+    // Unfortunately a string default must be provided here.
+    // The auth provider will fail to initially register if not provided,
+    // even when the custom "domain" field is provided because
+    // getAdapterDomain() returns null
+
+    return $this->domain ?: 'bugzilla.mozilla.org';
```

Comment:
This is frustrating because I'd like the DB value to be updated if the user updates the edit form.  The damn "providerDomain" field isn't updating in the DB upon save:

<img width="1502" alt="db" src="https://cloud.githubusercontent.com/assets/46655/25453315/0ef93814-2a8e-11e7-94a7-86931d1de4ef.png">

The custom property, however, does update.  I'm still working on this, but for now, this default value is needed to get the auth provider to register.